api-specifications.md

```diff
@@ -55,6 +52,26 @@ Each file and directory tree will be transferred individually.
 
 ---
 
+## Requests and Responses
+
+### Requests
+Clients will send HTTP request to server. Possible methods are:
+* GET
+* POST
+* DELETE
+
+The method to use is specified in each API entry point below
+
+### Response
+When client is not authenticaed (e.g. not logged in), server will return HTTP 301 and redirect user to log in page
+
+When client is authenticated, server will return 200 and a JSON object (specified below).
+* When success, the response will be `{"success": true, ...}`, where "`...`" contains extra information
+* When fail, the response will be `{"success": false, "message": "Error Message"}`. "`Error Message`" is defined in each "Error messages" sections below.
+* When server encounters an error, it will return 500. 
```

Comment:
When server encounters an internal error, it will return 500. In this case, the client should submit a bug report and report this to ngshare maintainers.